util/embedded-ldap/src/main/java/org/keycloak/util/ldap/LDAPEmbeddedServer.java

```diff
@@ -240,23 +265,65 @@ protected LdapServer createLdapServer() {
 
         ldapServer.setServiceName("DefaultLdapServer");
         ldapServer.setSearchBaseDn(this.baseDN);
+        // Tolerate plaintext LDAP connections from clients by default
+        ldapServer.setConfidentialityRequired(false);
 
         // Read the transports
         Transport ldap = new TcpTransport(this.bindHost, this.bindPort, 3, 50);
         ldapServer.addTransports( ldap );
-        if (enableSSL) {
-            Transport ldaps = new TcpTransport(this.bindHost, this.bindLdapsPort, 3, 50);
-            ldaps.setEnableSSL(true);
+        if (enableSSL || enableStartTLS) {
             ldapServer.setKeystoreFile(keystoreFile);
             ldapServer.setCertificatePassword(certPassword);
-            ldapServer.addTransports( ldaps );
+            // Require the LDAP server to accept only encrypted connections in the case LDAP over SSL / StartTLS operation mode was requested
+            ldapServer.setConfidentialityRequired(true);
```

Comment:
I am not sure if this change is correct. IMO the StartTLS is something, which client can use or not. So perhaps it should be still possible to connect through the "plain" connection even if StartTLS support or "enableSSL" support is enabled.

Reply:
Ok, fair enough. I will redo it differently yet & retest yet. Mmt

Reply:
Since StartTLS is [opportunistic encryption](https://en.wikipedia.org/wiki/Opportunistic_encryption) there is of course an option to fallback to insecure connection - it becomes a policy question if that is acceptable or not.
But from the point of StartTLS test it is of course important to make sure that StartTLS was really used :)

Reply:
Right. Originally thought, if StartTLS is requested, confidentiality should be enforced for all tests. But will change the PR to be more tolerant (still require confidentiality == true for StartTLS & LDAPS for this set of newly added LDAP tests), but tolerate plaintext connections for the existing tests already. Hopefully that's reasonable compromise (otherwise we might need to rewrite those failing tests to truly support StartTLS).

Reply:
+1 for require confidentiality==true just for the StartTLS tests (and eventually SSL tests if it is not additional headaches to cover). IMO other existing LDAP tests not related to StartTLS feature should not use StartTLS. Among other things, StartTLS connection is slower than the "plaintext" connection AFAIK.

Reply:
@mposolda @tsaarni Updated, should be ready for re-review again past the Travis run finished. The logic was changed to require confidentiality with StartTLS / LDAPS connections if and only if the StartTLS / LDAPS mode on LDAPEmbeddedServer was requested via the test annotation. Otherwise it will default to false (allow also unencrypted connections)